Upstream since 2025-02 but no release
---
From 066f88db28a8268981e9739fd01227bae233a89b Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Sun, 9 Feb 2025 16:16:10 -0800
Subject: [PATCH] add systemd user unit file

This is from Debian's vvmd source repo, I think it's useful to maintain
this upstream instead of in Debian so that other distros can use it
easily.
---
 data/meson.build     | 18 ++++++++++++++++++
 data/vvmd.service.in | 32 ++++++++++++++++++++++++++++++++
 meson.build          |  1 +
 3 files changed, 51 insertions(+)
 create mode 100644 data/meson.build
 create mode 100644 data/vvmd.service.in

diff --git a/data/meson.build b/data/meson.build
new file mode 100644
index 0000000..3d6c0c4
--- /dev/null
+++ b/data/meson.build
@@ -0,0 +1,18 @@
+# FIXME: make configurable with a meson option? GNOME Calls is a good example
+libdir = get_option('libdir')
+prefix = get_option('prefix')
+systemd_user_unit_dir =  prefix / libdir / 'systemd/user'
+
+service_conf = configuration_data()
+full_bindir = prefix / get_option('bindir')
+service_conf.set('bindir', full_bindir)
+
+# Systemd user service
+systemd_unit = 'vvmd.service'
+configure_file(
+  input : systemd_unit + '.in',
+  output : systemd_unit,
+  configuration : service_conf,
+  install : true,
+  install_dir : systemd_user_unit_dir,
+)
diff --git a/data/vvmd.service.in b/data/vvmd.service.in
new file mode 100644
index 0000000..cb3d90b
--- /dev/null
+++ b/data/vvmd.service.in
@@ -0,0 +1,32 @@
+[Unit]
+Description=Visual Voicemail Daemon
+After=ModemManager.service
+ConditionUser=!@system
+
+[Service]
+ExecStart=@bindir@/vvmd
+
+Restart=on-failure
+RestartSec=10s
+
+LockPersonality=yes
+MemoryDenyWriteExecute=yes
+NoNewPrivileges=yes
+RestrictNamespaces=yes
+SystemCallArchitectures=native
+SystemCallFilter=@system-service
+UMask=0077
+CapabilityBoundingSet=
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
+RestrictRealtime=yes
+PrivateDevices=yes
+PrivateMounts=yes
+PrivateTmp=yes
+PrivateUsers=yes
+ProtectClock=yes
+ProtectSystem=yes
+ProtectHostname=yes
+RestrictSUIDSGID=yes
+
+[Install]
+WantedBy=default.target
diff --git a/meson.build b/meson.build
index 847eef5..14d8773 100644
--- a/meson.build
+++ b/meson.build
@@ -139,3 +139,4 @@ vvmd = executable('vvmd',
 )
 
 subdir('unit')
+subdir('data')
-- 
GitLab

